server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -1043,6 +1134,11 @@ protected String contentType() {
 
     @Override
     protected void parseCreateField(ParseContext context) throws IOException {
+
+        if (this.script != null) {
+            throw new IllegalArgumentException("Cannot index data directly into scripted field");
```

Comment:
meta: we need to come up with a name that we all agree on for this functionality, and use that name consistently